TicTacToe/TicTacToe.py

```diff
@@ -0,0 +1,152 @@
+def get_board_size():
+    """ Board must be squared and and odd (with one center """
+    while True:
+        try:
+            print("Enter board Size (odd number)")
+            board_size = int(input())
+        except ValueError:
+            print("Board size must be a Number")
+            continue
+        if(board_size % 2 == 0 ):
+            print("Number must be odd")
+        # is odd number
+        else:
+            break
+    return board_size
+
+
+def init_board(board_size):
+    """ print blanked board with '0' in each cell """
+    board = [[0 for cell in range(0,board_size)] for row in range(0,board_size)]
+    return board
+
+
+def print_board_state(board):
+    """ Draw the board after each turn """
+    for row in board:
+        print(" -"*len(board))
+        print("|", end='')
+        for cell  in row:
+            if cell  == 0:
+                print(" ", end='')
+            else:
+                print(cell , end='')
+            print("|",end='')
+        print("")
+        print(" -"*len(board))
+
+
+def choose_player(board):
+    """ axiom Player X always starts """
+    sum_X_on_board = 0
+    sum_O_on_board = 0
+    for row in board:
+        sum_X_on_board += row.count("X")
+        sum_O_on_board += row.count("O")
+    # out of blanked cells is a Tie
+    if((sum_X_on_board+sum_O_on_board) == (len(board)*len(board))):
+        print("Game Done it's a Tie")
+        sign = "Tie"
+    elif(sum_X_on_board <= sum_O_on_board or sum_X_on_board == 0):
+        print("Player X turn:")
+        sign = "X"
+    else:
+        print("Player O turn:")
+        sign = "O"
+    return sign
+
+
+def play_random_move(board, sign):
+    """ choose cell randomly from all the empty places """
+    from random import randrange
+    x_axis = 0
+    y_axis = 0
+    empty_pointers = []
+    for row in board:
+        y_axis = 0
+        for cell in row:
+            if(cell == 0):
+                empty_pointers.append([x_axis,y_axis])
+            y_axis+=1
+        x_axis+=1
+    random_pointer_from_empty = randrange(0,len(empty_pointers))
+    # get x_axis and y_axis from the random_pointer and put there sign on board
+    board[empty_pointers[random_pointer_from_empty][0]][empty_pointers[random_pointer_from_empty][1]] = sign
+    return empty_pointers[random_pointer_from_empty]
+
+
+def check_winning_move(board, last_move):
+    """ Check only around the last move """
+    done = check_straight_winning(board, last_move)
+    if(not done):
```

Comment:
how is this working? you can win diagonally also if the move was not made on a corner or the center of the board.. i think you have a bug here